base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java

```diff
@@ -398,9 +400,17 @@ public void onReceived(RPCMessage message) {
                             VehicleType vehicleType = raiResponse.getVehicleType();
                             String systemSoftwareVersion = raiResponse.getSystemSoftwareVersion();
                             if (vehicleType != null || systemSoftwareVersion != null) {
-                                String address = session.getBluetoothMacAddress();
-                                if (address != null && !address.isEmpty()) {
-                                    saveVehicleType(address, vehicleType);
+                                List<TransportRecord> activeTransports = session.getActiveTransports();
+                                if (activeTransports != null) {
+                                    for (TransportRecord record: activeTransports) {
+                                        if (record.getType() == TransportType.BLUETOOTH) {
+                                            String address = record.getAddress();
+                                            if (address != null && !address.isEmpty()) {
+                                                saveVehicleType(address, vehicleType);
+                                            }
+                                            break;
```

Comment:
@YaroslavLutsenko maybe we need to move this break under `saveVehicleType`? I mean if BT transport has an empty address, probably we need to check another BT device? Or remove this break completely?

Reply:
@AKalinich-Luxoft I'm not sure that we can have several bluetooth transports per SdlSession. But I think we can remove this break.

Reply:
@YaroslavLutsenko I think it can be removed. If you are right - we will save vehicle type just once for one address. If not - it will be saved for both relevant transports which is also good :)